includes/AcquiaLiftLearnReport.inc

```diff
@@ -340,23 +309,67 @@ class AcquiaLiftLearnReport extends AcquiaLiftReportBase {
           'estimated_higher' => $this->formatReportNumber($upper, TRUE, 4),
           'estimated_lower' => $this->formatReportNumber($lower, TRUE, 4),
           'margin_error' => $this->formatReportNumber(($upper - $lower) / 2, TRUE, 4),
           'control' => $i === 0,
-          'confidence' => $confidence_formatted,
-          'lift_default' =>  $lift_default_formatted,
-          'lift_random' => 0,
+          'confidence' => self::NA_STRING,
+          'lift_default' =>  self::NA_STRING,
+          'lift_random' => self::NA_STRING,
         );
         $total_means += $mean;
         $i++;
       }
-
-      // Now our $total_means figure will allow us to calculate lift over random.
+      $control_variation = reset($data);
+      // Now we can do some comparisons of the data.
       $rand_mean = $total_means / $num_variations;
+      $choice_ids = array_keys($data);
+
       foreach ($data as $choice_id => &$values) {
+        // We impose a threshold of a minimum number of decisions before
+        // presenting a confidence value.
+        if ($values['count'] >= self::MIN_DECISION_THRESHOLD) {
+          // We need an array that holds all *but* this variation.
+          $other_variations = drupal_map_assoc($choice_ids);
+          unset($other_variations[$choice_id]);
+          // Kep track of how many other variations this variation beats.
+          $number_beaten = $overall_confidence = 0;
+          foreach ($other_variations as $var) {
+            $confidence = self::getConfidence(self::getTScore($values['count'], $data[$var]['count'], $values['mean'], $data[$var]['mean'], $values['variance'], $data[$var]['variance']), self::getDGFree($values['count'], $data[$var]['count'], $values['variance'], $data[$var]['variance']));
+            // If the comparison is against the control variation, then this confidence
+            // metric is what we display as "chance to beat control".
+            if ($var == $control_variation['option_id']) {
+              $values['confidence'] = $this->formatReportPercentage($confidence);
+            }
+            if ($confidence < ($this->confidence_measure / 100)) {
+              // No point doing anymore comparisons, this is not a winner.
+              break;
+            }
+            // If this is the lowest confidence value we've seen, then this is
+            // what we'll show as the overall confidence for the test, if this
+            // variation turns out to be the overall winner.
+            if (empty($overall_confidence) || $confidence < $overall_confidence) {
+              $overall_confidence = $confidence;
+            }
+            $number_beaten++;
+          }
+          if ($number_beaten === count($other_variations)) {
+            $this->interpretedResults['winner'] = array_search($choice_id, $choice_ids);
+            $this->interpretedResults['confidence'] = $overall_confidence;
+          }
+        }
```

Comment:
@ynx please have a look at https://github.com/acquia/acquia_lift/commit/b2616b0c756333a8c254e69f10a72972a80a34ba - hopefully this is more to your taste :) I'm still not totally happy with it but this code will be fairly short-lived (will be replaced by the new reporting api) so I don't want to spend too much time on it.